src/queues/mock.rs

```diff
@@ -0,0 +1,98 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at https://mozilla.org/MPL/2.0/.
+
+use chrono::Utc;
+
+use super::{
+    notification::{
+        Bounce, BounceSubtype, BounceType, BouncedRecipient, ComplainedRecipient, Complaint,
+        Delivery, Notification, NotificationType,
+    },
+    Factory, Incoming, Message, Outgoing, QueueError,
+};
+use settings::Settings;
+
+#[derive(Debug)]
+pub struct Queue<'s> {
+    id: &'s str,
+}
+
+impl<'s> Factory<'s> for Queue<'s> {
+    fn new(id: &'s str, _settings: &Settings) -> Queue<'s> {
+        Queue { id }
+    }
+}
+
+impl<'s> Incoming for Queue<'s> {
+    fn receive(&self) -> Result<Vec<Message>, QueueError> {
+        if self.id == "incoming" {
```

Comment:
It's a bit hackish, but doing this allows the tests to assert that the sanctity of incoming and outgoing queues is preserved.